core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -978,7 +979,7 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean) extends Abstra
       s"${KafkaConfig.AdvertisedListenersProp} protocols must be equal to or a subset of ${KafkaConfig.ListenersProp} protocols. " +
       s"Found ${advertisedListeners.keySet}. The valid options based on currently configured protocols are ${listeners.keySet}"
     )
-    require(interBrokerProtocolVersion.onOrAfter(ApiVersion(messageFormatVersion)),
+    require(interBrokerProtocolVersion >= messageFormatVersion,
       s"message.format.version $messageFormatVersion cannot be used when inter.broker.protocol.version is set to $interBrokerProtocolVersion")
```

Comment:
Good point, it used to do it for `interBrokerProtocolVersion` before this PR and now it does for both. I'll fix that.

I realised that it now sets `LogConfig.MessageFormatVersionProp` to `0.10.0-IV0` for this case too (instead of the user-provided String). Shall I change that too?